src/command_buffer.rs

```diff
@@ -134,67 +135,66 @@ impl CommandBuffer {
         unsafe {
             components.put(|ptr, ty| self.add_inner(ptr, ty));
         }
-        self.entities.push(EntityIndex {
+        self.components[first_component..].sort_unstable_by_key(|c| c.ty);
```

Comment:
Are there performance implications for pushing this sort out of `run_on` and into `insert`/`spawn`? The total size of a `CommandBuffer` is likely small, but I wonder if this has some degenerate cases.

Reply:
It's been a long time since this code was written, but the only difference I can think of is if you drop or `clear` the `CommandBuffer` without using it, which really shouldn't take place on a hot path. In normal use, every range of components is sorted exactly once regardless.

Reply:
Oh, I see how that works! That makes sense to me, cool.